plutus-contract/src/Plutus/Contract/Wallet.hs

```diff
@@ -115,22 +119,41 @@ getUnspentOutput = do
 -- | Partial transaction that can be balanced by the wallet backend.
 data ExportTx =
         ExportTx
-            { partialTx   :: C.Tx C.AlonzoEra -- ^ The transaction itself
-            , lookups     :: [ExportTxInput] -- ^ The tx outputs for all inputs spent by the partial tx
-            , signatories :: [Text] -- ^ Key(s) that we expect to be used for balancing & signing. (Advisory) See note [Keys in ExportT]
+            { partialTx           :: C.Tx C.AlonzoEra -- ^ The transaction itself
+            , lookups             :: [ExportTxInput] -- ^ The tx outputs for all inputs spent by the partial tx
+            , signatories         :: [Text] -- ^ Key(s) that we expect to be used for balancing & signing. (Advisory) See note [Keys in ExportT]
+            , missingValueBalance :: MissingValueBalance -- ^ Additional value that we expect the wallet to provide inputs / outputs for (excl. fees)
             }
     deriving stock (Generic, Typeable)
 
 data ExportTxInput = ExportTxInput{txIn :: C.TxIn, txOut :: C.TxOut C.AlonzoEra}
     deriving stock (Generic, Typeable)
     deriving anyclass (ToJSON)
 
+data MissingValueBalance = MissingValueBalance
+    { missingInputVal  :: C.Value -- ^ Total value to be provided by additional inputs
+    , missingOutputVal :: C.Value -- ^ Total value to paid to additional outputs
+    }
+    deriving stock (Generic, Typeable)
+
+instance ToJSON MissingValueBalance where
+    toJSON MissingValueBalance{missingInputVal, missingOutputVal} =
+        object ["inputs" .= toJSON missingInputVal, "outputs" .= toJSON missingOutputVal]
+
+instance FromJSON MissingValueBalance where
+    parseJSON (Object v) =
+        MissingValueBalance
+            <$> v .: "inputs"
+            <*> v .: "outputs"
+    parseJSON _ = fail "Expected Object"
+
 instance ToJSON ExportTx where
-    toJSON ExportTx{partialTx, lookups, signatories} =
+    toJSON ExportTx{partialTx, lookups, signatories, missingValueBalance} =
         object
             [ "transaction" .= toJSON (C.serialiseToTextEnvelope Nothing partialTx)
             , "inputs"      .= toJSON lookups
             , "signatories" .= toJSON signatories
+            , "missing-value" .= toJSON missingValueBalance
```

Comment:
I think this should be `"missingValue"`, I don't see dashes used elsewhere in our JSON.